mods/bucket/init.lua

```diff
@@ -123,6 +123,7 @@ minetest.register_craftitem("bucket:bucket_empty", {
 				return
 			end
 
+			minetest.dig_node(pointed_thing.under)
 			minetest.add_node(pointed_thing.under, {name="air"})
```

Comment:
minetest.dig_node already removes the node, it's not necessary to set "air", is it?